lucene/benchmark/src/java/org/apache/lucene/benchmark/byTask/feeds/DirContentSource.java

```diff
@@ -0,0 +1,270 @@
+        String b = _b.toString();
+        int diff = a.length() - b.length();
+
+        if (diff > 0) {
+          while (diff-- > 0) {
+            b = "0" + b;
+          }
+        } else if (diff < 0) {
+          diff = -diff;
+          while (diff-- > 0) {
+            a = "0" + a;
+          }
+        }
+
+        /* note it's reversed because we're going to push,
+           which reverses again */
+        return b.compareTo(a);
+      }
+    }
+
+    int count = 0;
+
+    Stack<Path> stack = new Stack<>();
+
+    /* this seems silly ... there must be a better way ...
+       not that this is good, but can it matter? */
+
+    Comparator c = new Comparator();
+
+    public Iterator(Path f) throws IOException {
+      push(f);
+    }
+
+    void find() throws IOException {
+      if (stack.empty()) {
+        return;
+      }
+      if (!Files.isDirectory(stack.peek())) {
+        return;
+      }
+      Path f = stack.pop();
+      push(f);
+    }
+
+    void push(Path f) throws IOException {
+      try (DirectoryStream<Path> stream = Files.newDirectoryStream(f)) {
+        List<Path> found = new ArrayList<>();
+        for (Path p : stream) {
+          if (Files.isDirectory(p)) {
+            found.add(p);
+          }
+        }
+        push(found.toArray(new Path[found.size()]));
+      }
+      try (DirectoryStream<Path> stream = Files.newDirectoryStream(f, "*.txt")) {
+        List<Path> found = new ArrayList<>();
+        for (Path p : stream) {
+          found.add(p);
+        }
+        push(found.toArray(new Path[found.size()]));
+      }
+      find();
+    }
+
+    void push(Path[] files) {
+      Arrays.sort(files, c);
+      for(int i = 0; i < files.length; i++) {
+        // System.err.println("push " + files[i]);
+        stack.push(files[i]);
+      }
+    }
+
+    public int getCount(){
+      return count;
+    }
+
+    @Override
+    public boolean hasNext() {
+      return stack.size() > 0;
+    }
+    
+    @Override
+    public Path next() {
+      assert hasNext();
+      count++;
+      Path object = stack.pop();
+      // System.err.println("pop " + object);
+      try {
+        find();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+      return object;
+    }
+
+    @Override
+    public void remove() {
+      throw new RuntimeException("cannot");
+    }
+
+  }
+  
+  private ThreadLocal<DateFormatInfo> dateFormat = new ThreadLocal<>();
+  private Path dataDir = null;
+  private int iteration = 0;
+  private Iterator inputFiles = null;
+
+  // get/initiate a thread-local simple date format (must do so 
+  // because SimpleDateFormat is not thread-safe).
+  private DateFormatInfo getDateFormatInfo() {
+    DateFormatInfo dfi = dateFormat.get();
+    if (dfi == null) {
+      dfi = new DateFormatInfo();
+      dfi.pos = new ParsePosition(0);
+      // date format: 30-MAR-1987 14:22:36.87
+      dfi.df = new SimpleDateFormat("dd-MMM-yyyy kk:mm:ss.SSS", Locale.ENGLISH);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.